tests/install/test.py

```diff
@@ -1,19 +1,66 @@
+import atexit
 import logging
 import os
 import pytest
+import tempfile
 import time
 
 from lib import commands, installer, pxe
-from lib.common import safe_split, wait_for
+from lib.common import safe_split, wait_for, callable_marker
 from lib.pif import PIF
 from lib.pool import Pool
 
 from data import NETWORKS
 assert "MGMT" in NETWORKS
 
+from data import HOST_DEFAULT_PASSWORD_HASH
+from data import ISO_IMAGES
+
+def clean_files_on_pxe(mac_address):
+    pxe.server_remove_config(mac_address)
+
+def setup_pxe_boot(vm, mac_address, version):
+    with tempfile.TemporaryDirectory(suffix=mac_address) as tmp_local_path:
+        logging.info('Generate answerfile.xml for {}'.format(mac_address))
+
+        hdd = 'nvme0n1' if vm.is_uefi else 'sda'
+        encrypted_password = HOST_DEFAULT_PASSWORD_HASH
+        try:
+            installer = ISO_IMAGES[version]['pxe-url']
+        except KeyError:
+           pytest.skip(f"cannot found pxe-url for {version}")
+        pxe_addr = pxe.PXE_CONFIG_SERVER
+
+        with open(f'{tmp_local_path}/answerfile.xml', 'w') as answerfile:
+            answerfile.write(f"""<?xml version="1.0"?>
+<installation>
+    <keymap>fr</keymap>
+    <primary-disk>{hdd}</primary-disk>
+    <guest-disk>{hdd}</guest-disk>
+    <root-password type="hash">{encrypted_password}</root-password>
+    <source type="url">{installer}</source>
+    <admin-interface name="eth0" proto="dhcp" />
+    <timezone>Europe/Paris</timezone>
+    <script stage="filesystem-populated" type="url">
+        http://{pxe_addr}/configs/presets/scripts/filesystem-populated.py
+    </script>
+</installation>
+            """)
+
+        logging.info('Generate boot.conf for {}'.format(mac_address))
+        pxe.generate_boot_conf(tmp_local_path, installer)
+
+        logging.info('Copy files to {}'.format(pxe_addr))
+        pxe.server_push_config(mac_address, tmp_local_path)
+
+        # FIXME: clean_files_on_pxe has been commented on lib/pxe.py
+        # Note: even commented the files are cleaned up on the PXE server???
+        logging.info('Register cleanup files for PXE')
+        atexit.register(lambda: clean_files_on_pxe(mac_address))
+
 @pytest.mark.dependency()
 class TestNested:
-    @pytest.mark.parametrize("source_type", ("local", "netinstall"))
+    @pytest.mark.parametrize("source_type", ("local", "netinstall", "pxe"))
```

Comment:
I have mixed feelings about using this parameter.  `source_type` is rather meant for "where to take package repo from".  When we're booting with PXE the package source will have to be `netinstall`.
I think it would make sense to have something like
```
@pytest.mark.parametrize("boot,source_type", (
    ("iso", "iso"),
    ("iso", "net"),
    ("pxe", "net"),
))
```